farm/contract/lib.rs

```diff
@@ -85,6 +85,12 @@ mod farm {
             if reward_tokens.contains(&pool_id) {
                 return Err(FarmError::RewardTokenIsPoolToken);
             }
+            let mut _reward_tokens: Mapping<TokenId, ()> = Mapping::new();
```

Comment:
It's regular `Mapping` that writes to the underlying storage as there's no `HashMap` available in ink! and I didn't want to check all permutations of `TokenId`s. 

I was about to say that we don't write anything actually b/c this is just temporary for the duration of the constructor but that's not true. I'll add the logic to remove all keys at the end of the constructor - this way we really don't write any additional data to storage and the cost is paid once, by the creator of the farm.

Reply:
I think that unnecessarily complicates that check, as you write to storage and then clean it instead of just having two loops that make no more than 45 comparisons total (just once in the constructor).

Reply:
I still think readability is much better with the current version than two loops of iteration but I'll do what you suggest.

Reply:
I also like 2 nested loops much more...

Reply:
Done in [Replace Mapping::insert with loops](https://github.com/Cardinal-Cryptography/common-amm/pull/86/commits/f1db02cb0dd3aa7cd777ae7e7273f2008cb72b98).